src/domain/community/community/community.resolver.mutations.ts

```diff
@@ -189,6 +186,51 @@ export class CommunityResolverMutations {
     return savedApplication;
   }
 
+  @UseGuards(GraphqlGuard)
+  @Mutation(() => ICommunity, {
+    description: 'Join the specified Community as a member.',
+  })
+  @Profiling.api
+  async joinCommunity(
+    @CurrentUser() agentInfo: AgentInfo,
+    @Args('joinCommunityData') joiningData: CommunityJoinInput
+  ): Promise<ICommunity> {
+    const community = await this.communityService.getCommunityOrFail(
+      joiningData.communityID
+    );
+
+    const user = await this.userService.getUserOrFail(joiningData.userID);
+    // A user can only apply for themselves
+    if (user.id !== agentInfo.userID) {
+      throw new NotSupportedException(
+        `Applications for community membership must be made by the current user: ${joiningData.userID} - ${user.nameID}`,
+        LogContext.COMMUNITY
+      );
+    }
+
+    await this.authorizationService.grantAccessOrFail(
+      agentInfo,
+      community.authorization,
+      AuthorizationPrivilege.COMMUNITY_APPLY,
+      `create application community: ${community.displayName}`
+    );
+
+    // Todo: notification for new community membership
+    // const payload =
+    //   await this.notificationsPayloadBuilder.buildApplicationCreatedNotificationPayload(
+    //     agentInfo.userID,
+    //     joiningData.userID,
+    //     community
+    //   );
+
+    // this.notificationsClient.emit<number>(
```

Comment:
This will disable the notification for that event. Isn't it better to leave in the current one and address it at a later stage?

Reply:
This needs to be a totally different notification, and probably be "new community member". The application notification would not be appropriate.